### DISCUSSION TITLE
Keyboard unresponsive after login

### DISCUSSION THREAD

**Discussion by tinkerpunktheprol** [i3/i3#5328] [Q&A] [answered]
I posted this on the EndeavourOS forums yesterday and haven't received any help yet, so I figured I'd ask you all to take a look if you could. Normally I'd be a little more patient, but I keep trying to type on my laptop keyboard instead of my USB one, and it's really starting to piss me off, lol. I'm just going to paste my post from over there, I hope that is ok.

<a id="org670046d"></a>

# Issue:

Keys on built-in laptop keyboard are not working after login to i3

<a id="orga30d4a3"></a>

# Logs:

[inxi](https://0x0.st/onVz.txt)
[journalctl](https://0x0.st/onV-.txt)
[cat /var/log/Xorg.0.log | grep fail | eos-sendlog](https://0x0.st/onVN.txt) - the EOS tutorial post for sharing logs made me nervous about doing this manually, so I added the `grep fail`. If this is not enough information, please let me know, or if I need to share any other logs.

<a id="orge53c097"></a>

# Cause:

While trying to get my brightness controls working on my laptop, I found [this](https://itsfoss.com/fix-brightness-ubuntu-1310/) tutorial and did the following:

After finding the name of my backlight by running `ls /sys/class/backlight/`, I then edited `/usr/share/X11/xorg.conf.d/10-amdgpu.conf/`:

Here is that file prior to me editing it:

    Section "OutputClass"
    	Identifier "AMDgpu"
    	MatchDriver "amdgpu"
    	Driver "amdgpu"
    EndSection

And what I added:

    Section "Device"
            Identifier  "card0"
            Driver      "amdgpu"
            Option      "Backlight"  "amdgpu_bl1"
            BusID       "PCI:0:2:0"

Just like that. And then I logged out, as the article says. I was immediatly greeted with a black screen and a blinking cursor. As you may have noticed&#x2026; I failed to include the `EndSection`

I eventually found my mistake and fixed it via console. I added that last element, and rebooted again, but no cigar. I booted back to console and totally removed the code I added and it booted up fine. But&#x2026;my keyboard is no longer working. Like, no keys at all. The power button seems to be the only thing functioning. I also noticed that my trackpad is enabled again, which I wrote a script to disable and added to i3wm config to trigger on startup. This was working fine prior to this event.

All usb devices are working. I am currently typing this on the laptop with a usb keyboard.

<a id="orgac78ca9"></a>

# My thoughts

I have no clue how PCI id&rsquo;s are allocated, but I was wondering whether or not `PCI:0:2:0` belongs to a specific device, and maybe that device was my keyboard, and I took it&rsquo;s id &ldquo;slot&rdquo; with the backlight.

[lspci | eos-sendlog](https://0x0.st/onVY.csv)

    00:02.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Renoir PCIe Dummy Host Bridge

That is the closest thing I could find to the id I was told to use. Again, not sure how PCI device alocation works.

Please let me know if there is any more information needed. Thanks ahead of time to anyone that can offer some help.

**Comment by tinkerpunktheprol** [comment, marked as answer]
I figured it out. Somehow in the process of re-configuring the brightness control, the xinput id's got redistributed and the keyboard got id 17, which was my touchpad's id before. So now my script was disabling my keyboard instead.

I still have no idea why this happened, though. Could someone please point me in the direction of some search terms to use to learn more? The link between PCI id's and xinput makes sense, but they seem to have different configuration processes, at least from my beginners perspective.

**Reply by vincentbernat** [reply]
I don't think there is any link between XInput IDs and PCI IDs. XInput IDs are allocated in the order the X server sees the devices which may be quite random. You can use the device name instead.

**Reply by tinkerpunktheprol** [reply]
That's a good idea, thank you. Does Xinput change ID's at certain intervals?
